addons/account/models/res_partner_bank.py

```diff
@@ -1,15 +1,217 @@
+                raise werkzeug.exceptions.HTTPException(description='Cannot convert into barcode.')
+            return image_data_uri(base64.b64encode(barcode))
+        return None
+
+    @api.model
+    def _get_available_qr_methods(self):
+        """ Returns the QR-code generation methods that are available on this db,
+        in the form of a list of (code, name, sequence) elements, where
+        'code' is a unique string identifier, 'name' the name to display
+        to the user to designate the method, and 'sequence' is a positive integer
+        indicating the order in which those mehtods need to be checked, to avoid
+        shadowing between them (lower sequence means more prioritary).
+        """
+        return []
+
+    @api.model
+    def get_available_qr_methods_in_sequence(self):
+        """ Same as _get_available_qr_methods but without returning the sequence,
+        and using it directly to order the returned list.
+        """
+        all_available = self._get_available_qr_methods()
+        all_available.sort(key=lambda x: x[2])
+        return [(code, name) for (code, name, sequence) in all_available]
+
+    def _eligible_for_qr_code(self, qr_method, debtor_partner, currency, raises_error=True):
+        """ Tells whether or not the criteria to apply QR-generation
+        method qr_method are met for a payment on this account, in the
+        given currency, by debtor_partner. This does not impeach generation errors,
+        it only checks that this type of QR-code *should be* possible to generate.
+        Consistency of the required field needs then to be checked by _check_for_qr_code_errors().
+        """
+        return False
+
+    def _check_for_qr_code_errors(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
+        """ Checks the data before generating a QR-code for the specified qr_method
+        (this method must have been checked for eligbility by _eligible_for_qr_code() first).
+
+        Returns None if no error was found, or a string describing the first error encountered
+        so that it can be reported to the user.
+        """
+        return None
+
+    def create(self, vals_list):
+        # EXTENDS base res.partner.bank
+        res = super().create(vals_list)
+        if not self.env.context.get('tracking_disable'):
+            for account in res:
+                msg = _("Bank Account %s created", account._get_html_link(title=f"#{account.id}"))
+                account.partner_id._message_log(body=msg)
+        return res
+
+    def write(self, vals):
+        # EXTENDS base res.partner.bank
+        # Track and log changes to partner_id, heavily inspired from account_move
+        if not self.env.context.get('tracking_disable'):
+            # Get all tracked fields (without related fields because these fields must be manage on their own model)
+            tracking_fields = []
+            for value in vals:
```

Comment:
That's a bit confusing as usually `value` denotes the value of a dict, but here it denotes the key (a field name).